### DISCUSSION TITLE
Payment Brick unmounting on any position displacement

### DISCUSSION THREAD

**Discussion by AndAccioly** [mercadopago/sdk-js#207] [Q&A] [answered]
Good night, please I need some help. I am doing a web store and I am using mercadopago/sdk-react v0.0.19 for payment. But, whenever I resize the page or roll down the page the Payment Brick component renderizes again (it vanishes and then mount it again).

It is worse on mobile because when I click on card field to write my card number and the keyboard goes up, the displacement of the Payment component just make it unmount and mount again.

The brick component is just as it is below.

<Payment
      initialization={initialization}
      customization={customization}
      onSubmit={onSubmit}
      onReady={onReady}
      onError={onError}
  />

I hope anyone can help me!

**Comment by AleFossati** [comment, marked as answer]
Hello @AndAccioly.

I have tried a new implementation of the Payment brick and it seems to me that it is working fine even resizing the page multiple times.
Here's the [link of the public implementation](https://codesandbox.io/p/sandbox/lingering-night-fz7p47?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clyhjxlg500063b6kwwn06w07%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clyhjxlg400023b6k2tf6vsen%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clyhjxlg400033b6kde1fvh5o%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clyhjxlg400053b6k7njyx3ed%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B51.900492365461616%252C48.099507634538384%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clyhjxlg400023b6k2tf6vsen%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyhjxlg400013b6kyuxm6wms%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%257D%255D%252C%2522id%2522%253A%2522clyhjxlg400023b6k2tf6vsen%2522%252C%2522activeTabId%2522%253A%2522clyhjxlg400013b6kyuxm6wms%2522%257D%252C%2522clyhjxlg400053b6k7njyx3ed%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyhjxlg400043b6kjbji6pro%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clyhjxlg400053b6k7njyx3ed%2522%252C%2522activeTabId%2522%253A%2522clyhjxlg400043b6kjbji6pro%2522%257D%252C%2522clyhjxlg400033b6kde1fvh5o%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clyhjxlg400033b6kde1fvh5o%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D), so you can check it out.

The re-renders is a super common issue when using react, but it is mostly related to the state management of your application.
Maybe it's something related to some `useState` that you have in your component, or even a parent component.
Here's some guides that we recommend the reading to understand it better.

[React.dev - Memo](https://react.dev/reference/react/memo)
[Medium - React Components rendered twice](https://andreasheissenberger.medium.com/react-components-render-twice-any-way-to-fix-this-91cf23961625)
[Medium - Wait, you’re not using <React.StrictMode>?!](https://medium.com/nmc-techblog/wait-youre-not-using-react-strictmode-a9713927a33b)

If you have any other question, let me know.

**Comment by AndAccioly** [comment]
Thank you so much taking some of you time to answer me, and you were absolutely right. For what you said and the example code, I could test some stuff and find there is a hook that I made to check width and height to check screen size so I can choose to render between mobile and desktop different views, and that was causing the re-rendering.

Thank you so much again! You saved my week xD
